### ISSUE TITLE
Rigid body transform registration

### ISSUE THREAD

**Issue by damaggu** [stnava/ANTs#399]
Hi, 

I work with preclinical scanners and thus it happens, that the orientation of my image is not the same as the one of my template (axis not the same). 

I find that ANTs won't rotate my image sufficiently in order to change it's orientation completely.

Is there a way to achieve that or is there a fundamental limit on the degrees of rotation in ants?

Thanks.

**Comment by cookpa** [comment]
PermuteFlipImageOrientationAxes

**Comment by damaggu** [comment]
Thanks cookpa, but I'd like to automate that. Is that possible?

**Comment by TheChymera** [comment]
Yes, I would also have thought that Rigid or Symilarity transformations would be able to do this.  `PermuteFlipImageOrientationAxes` seems to do the same thing as FSL's `fslswapdim`. Ideally we would want our pipeline to handle this automatically. We've figured it out the hard way, but not every user of a preclinical scanner knows how to flip data matrices and affine transforms around. Actually almost none.

**Comment by cookpa** [comment]
Are the input images consistent? If so, can't you figure out the necessary transform one time? 

If the input images are not consistent, it will be more difficult to automate. You can try using `antsAI` to try a bunch of different initial orientations, but you might need to tweak the settings to try a large search space.

**Comment by TheChymera** [comment]
Sadly, they are not consistent. Preclinical scanners and their sequences tend to be a lot more customized and standardized than for human MRI.

Is there any way to set the initial orientations for `antsRegistration`? This is the command we prefer to use because it gives us the greatest control and transparency  for the registration parameters.

**Comment by stnava** [comment]
see ?antsRegistration in antsr or antsRegistration --help for how to
initialize

?invariantImageSimilarity could help too ... the example shows how to use
them together:

> library(ANTsR)

> ?invariantImageSimilarity

>     fi<-antsImageRead( getANTsRData("r16") )

>      mi<-antsImageRead( getANTsRData("r64") )

>      mival<-invariantImageSimilarity( fi, mi, thetas = c(0,10,20) )

>      mapped = antsApplyTransforms( fi, mi, transformlist=mival[[2]] )

>      areg = antsRegistration( fi, mi, typeofTransform="Affine",

+        initialTransform=mival[[2]] )


and:


https://github.com/stnava/ANTsR/wiki/Multistart-optimization-for-challenging-registration-problems


brian

**Comment by spinicist** [comment]
Hi @damaggu and @TheChymera,

I think there are possibly two separate issues here, both of which I have dealt with. @damaggu's problem is that the definition of 'coronal' is ambiguous for non-primates. Most rodent templates use the human/primate definition of coronal so that, for example, the labels in FSLView appear correct. Some stick with the orientation rodents are actually scanned in.

The way I deal with this is to set the initial orientation for `antsRegistration` using the `-r` option. I wrote a small utility to construct itkTransform files for basic rotation around axes - usually a 90 degree rotation about X and 180 about Z does the trick. I then pass that file into `antsRegistration` and it generally works (while writing this post I have realised I should experiment with the `-q` option which would let me still specify CoG for `-r`). This is the `qiaffine` tool in https://github.com/spinicist/QUIT/tree/development, if you are interested.

@TheChymera's problem sounds more like the orientations being inconsistent between sequences. What platform are you on? If it's Agilent/Varian then with the help of an Agilent engineer I wrote a conversion tool from Agilent->Nifti that is very careful to get the orientations correct. At least with my own projects, I haven't found a bug in it (yet). That's in: https://github.com/spinicist/AgilentTools
